src/bot.ts

```diff
@@ -435,15 +435,32 @@ export class DiscordBot {
         return dbEmoji.MxcUrl;
     }
 
-    public async GetRoomIdsFromGuild(guild: string): Promise<string[]> {
-        const rooms = await this.bridge.getRoomStore().getEntriesByRemoteRoomData({
-            discord_guild: guild,
-        });
-        if (rooms.length === 0) {
-            log.verbose(`Couldn't find room(s) for guild id:${guild}.`);
-            throw new Error("Room(s) not found.");
+    public async GetRoomIdsFromGuild(guild: Discord.Guild, member?: Discord.GuildMember): Promise<string[]> {
+        if (member) {
+            let rooms: string[] = [];
+            await Util.AsyncForEach(guild.channels.array(), async (channel) => {
+                if (channel.type !== "text" || !channel.members.has(member.id)) {
+                    return;
+                }
+                try {
+                    rooms = rooms.concat(await this.channelSync.GetRoomIdsFromChannel(channel));
```

Comment:
Wait, this can throw?

Reply:
yes, if there are no rooms in `this.channelSync.GetRoomIdsFromChannel`